### ISSUE TITLE
`tmt plans lint` does have hidden local cache in /home/user directory, which contents are not updated when the command is called

### ISSUE THREAD

**Issue by zdohnal** [teemtee/tmt#3097]
Hi,

I was checking whether my plans are ok on the component, so I used 'tmt plans lint', which showed me an issue:

```
fail P005 remote fmf id in "golang-github-openprinting-goipp" is invalid, No tree found in repo 'https://src.fedoraproject.org/tests/golang-github-openprinting-goipp.git', missing an '.fmf' directory?
```

so I went to the mentioned repo and added the missing dir.

However the test kept failing, so I started to debug tmt linting. During that action I found the remote ref dir tree is downloaded into ~/.cache/fmf, which is not automatically swept, so files keep piling up there.

Additionally, the lint started to pass in the end, so it looked like ```tmt plans lint``` was not downloading data every time I called the command, but only after some time, like the data had some timer...

$ rpm -q tmt
tmt-1.34.0-1.fc39.noarch

**Comment by psss** [comment]
Yes, there is a cache implemented in `fmf` to speed-up some use cases. The default expiration is 20 minutes.

https://github.com/teemtee/fmf/blob/39af408279b305bd8744e1a74c4ad09958394587/fmf/utils.py#L46-L47

I guess we have several options how to handle this:

* turn off the cache for the `tmt plan` commands (would slow down the exploration)
* provide a simple way to clean up the cache (directly from the `tmt plan` command)
* just document the cache existence and that `fmf cleanup` can be used to clean it up

@zdohnal, what do you think would be the best approach here?

**Comment by happz** [comment]
> I guess we have several options how to handle this:
> 
> * turn off the cache for the `tmt plan` commands (would slow down the exploration)
> * provide a simple way to clean up the cache (directly from the `tmt plan` command)
> * just document the cache existence and that `fmf cleanup` can be used to clean it up

Another option: make tmt aware of the cache, and let tmt manage it. `tmt run` should be able to log which repositories and git refs were used for testing, what has been cloned and so on. A hidden cache does not make this task easier.

**Comment by psss** [comment]
Might also be a duplicate of:

* #2918

**Comment by zdohnal** [comment]
Tbh IMO `tmt plans lint` does not have to leave any cache after its run - it is not a thing which is expected to be run repeatedly and there is no requirement for performance here. But for other commands I agree with @happz .

**Comment by happz** [comment]
As discussed, the most tempting option is to implement a cache component inside tmt that would be responsible for this part of tmt pipeline. fmf would hand over control of its case to tmt, and tmt would be able to reliably work with git repositories.
